crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -2084,8 +2084,11 @@ where
             }) => {
                 self.visit_boolean_test(test);
 
-                if flake8_type_checking::helpers::is_type_checking_block(&self.semantic_model, test)
-                {
+                if analyze::typing::is_type_checking_block(&self.semantic_model, test) {
```

Comment:
Wishlist: It would be nice if we could cache the `resolve_call_path` calls for as long as it is the same scope

Reply:
I agree...